source/mir/math/sum.d

```diff
@@ -190,44 +190,48 @@ nothrow @nogc unittest
 
 /// Moving mean
 version(mir_test)
+@safe pure nothrow @nogc
 unittest
 {
-    import mir.ndslice.topology: linspace;
     import mir.math.sum;
-    import mir.array.allocation: array;
+    import mir.ndslice.allocation: rcslice;
+    import mir.ndslice.topology: linspace;
 
-    class MovingAverage
+    struct MovingAverage(T, Summation summation)
     {
-        Summator!(double, Summation.precise) summator;
-        double[] circularBuffer;
+        import mir.math.stat: MeanAccumulator;
+        import mir.ndslice.slice: Slice;
+        import mir.rc.array: RCI;
+
+        MeanAccumulator!(T, summation) meanAccumulator;
+        Slice!(RCI!T) circularBuffer;
```

Comment:
it should be `double[]`, you may want to add `@disable this(this)`

Reply:
Happy to make this change, but I don't really understand the motivation for using `double[]` here, as opposed to a Slice, either just a simple contiguous one or the reference-counted one I used. 

Honestly, I expected that making the change would have led to a gc allocation and was a little surprised when it didn't.

Reply:
A `double[]` doesn't really mean a GC-allocated memory. It is just a slice of memory. The idea is to allow a user to reuse existing memory. For example, it may be a stack-allocated memory and fallback to a manually allocated memory in case of a large size.